File: Java/convert_array_into_2d_array_with_conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        List<List<Integer>> rows = new ArrayList<>();
        rows.add(new ArrayList<>());

        for(int i = 0; i < nums.length; i++) {
            Boolean found = false;
            for(int j = 0; j < rows.size(); j++) {
                if(!rows.get(j).contains(nums[i])){
                    rows.get(j).add(nums[i]);
                    found = true;
                    break;
                }
            }

            if(!found){
                List<Integer> aux = new ArrayList<>();
                aux.add(nums[i]);
                rows.add(aux);
            }
        }

        return rows;
    }
}

File: Python/convert_array_into_2d_array_with_conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        rows = []
        rows.append([])

        for num in nums: 
            found = False

            for i in range(len(rows)):
                if(not num in rows[i]):
                    rows[i].append(num)
                    found = True
                    break

            if(not found):
                aux = []
                aux.append(num)
                rows.append(aux)
        
        return rows
            